lib/ansible/modules/windows/win_domain_object_info.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Shachaf Goldstein <shachaf.gold@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_domain_object_info
+version_added: '2.10'
+short_description: retrieve  info regarding domain objects of all kinds
+description:
+   - With the module you can retrieve and output detailed information about all kinds of AD objects.
+options:
+  filter:
+    description:
+      - Powershell style filter.
+    type: str
+  identity:
+    description:
+      - Identity parameter used to find the Object in Active Directory.
+      - This value can be in the forms C(Distinguished Name), C(objectGUID).
+    type: str
+  include_deleted_objects:
+    description:
+      - Should deleted objects be included.
+    type: bool
+    default: false
+  ldap_filter:
+    description:
+      - LDAP style filter.
+    type: str
+  search_scope:
+    description:
+      - The scope of the search.
+      - A Base query searches only the current path or object.
+      - A OneLevel query searches the immediate children of that path or object.
+      - A Subtree query searches the current path or object and all children of that path or object.
+    type: str
+    default: Subtree
```

Comment:
Due to this not being a valid parameter with `-Identity` we should remove the default